src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component, Element, Prop, Watch, h } from '@stencil/core';
+
+const VALID_STATES = ['', 'success', 'warning', 'danger'];
+const VALID_SIZES = ['', 'sm', 'md', 'lg', 'xl'];
```

Comment:
Don't accept empty string as a valid value. Use the default value instead. If you set the property as optional, do this as this: 
```
@Prop() size? = "md"; //Or whatever the default value is
```